test/jdk/javax/swing/JTableHeader/TableHeaderBorderPositionTest.java

```diff
@@ -65,64 +64,40 @@ private static void test() {
         Dimension size = header.getPreferredSize();
         header.setSize(size);
 
-        int w = (int)Math.ceil(SCALE * size.width);
-        int h = (int)Math.ceil(SCALE * size.height);
-
-        BufferedImage imgHeader = new BufferedImage(w, h,
+        BufferedImage bufferedImage = new BufferedImage(WIDTH, HEIGHT,
                 BufferedImage.TYPE_INT_RGB);
-        Graphics2D g2d = imgHeader.createGraphics();
+        Graphics2D g2d = bufferedImage.createGraphics();
         g2d.scale(SCALE, SCALE);
-        try {
-            header.paint(g2d);
-        } finally {
-            g2d.dispose();
-        }
 
-        w = (int)Math.ceil(SCALE * WIDTH);
-        h = (int)Math.ceil(SCALE * HEIGHT);
-        BufferedImage imgData = new BufferedImage(w, h,
-                BufferedImage.TYPE_INT_RGB);
-        g2d = imgData.createGraphics();
-        g2d.scale(SCALE, SCALE);
         try {
+            header.paint(g2d);
+            g2d.translate(0, header.getHeight());
             table.paint(g2d);
         } finally {
```

Comment:
Awesome! I didn't think about it. If it works, it makes verification easier. When you read the code, you clearly see the borders of the header and grid align.

I expected the table would paint both its header and its contents, but it may require validation which can't happen until the table is added to a heavyweight container such as a frame.